src/app/Topology/GraphView/CustomNode.tsx

```diff
@@ -0,0 +1,154 @@
+import { css } from '@patternfly/react-styles';
+import {
+  DefaultNode,
+  DEFAULT_LAYER,
+  EllipseAnchor,
+  Layer,
+  Node,
+  observer,
+  ScaleDetailsLevel,
+  TOP_LAYER,
+  useAnchor,
+  useHover,
+  WithContextMenuProps,
+  WithDragNodeProps,
+  WithSelectionProps,
+} from '@patternfly/react-topology';
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import { getStatusTargetNode, isTargetMatched, nodeTypeToAbbr, useSearchExpression } from '../Shared/utils';
+import { TargetNode } from '../typings';
+import { RESOURCE_NAME_TRUNCATE_LENGTH } from './UtilsFactory';
+
+export const NODE_ICON_PADDING = 5;
+
+export const renderIcon = (data: TargetNode, element: Node, useAlt: boolean): React.ReactNode => {
+  const { width, height } = element.getDimensions();
+
+  const contentSize = Math.min(width, height) - NODE_ICON_PADDING * 2;
+  const mainContentSize = contentSize * (useAlt ? 0.5 : 0.8);
+  const [cx, cy] = [width / 2, height / 2];
+  const [trueCx, trueCy] = [cx - mainContentSize / 2, cy - mainContentSize / 2];
+
+  return (
+    <>
+      <circle cx={cx} cy={cy} r={contentSize / 2} fill="var(--pf-global--palette--white)" />
+      {useAlt ? (
+        <g transform={`translate(${trueCx}, ${trueCy})`}>
+          <ContainerNodeIcon width={mainContentSize} height={mainContentSize} />
+        </g>
+      ) : (
+        <image x={trueCx} y={trueCy} width={mainContentSize} height={mainContentSize} xlinkHref={openjdkSvg} />
+      )}
+    </>
+  );
+};
+
+export interface CustomNodeProps extends Partial<WithSelectionProps & WithDragNodeProps & WithContextMenuProps> {
+  element: Node;
+}
+
+export const NODE_BADGE_COLOR = 'var(--pf-global--palette--blue-500)';
+
+const CustomNode: React.FC<CustomNodeProps> = ({
+  element,
+  onSelect,
+  selected,
+  dragNodeRef,
+  contextMenuOpen,
+  onContextMenu,
+  ...props
+}) => {
+  useAnchor(EllipseAnchor); // For edges
+  const [hover, hoverRef] = useHover(200, 200);
+  const [expression] = useSearchExpression();
+
+  const displayOptions = useSelector((state: RootState) => state.topologyConfigs.displayOptions);
+  const { badge: showBadge, connectionUrl: showConnectUrl, icon: showIcon, status: showStatus } = displayOptions.show;
+
+  const detailsLevel = element.getController().getGraph().getDetailsLevel();
+  const labelIcon = React.useMemo(() => <img src={cryostatSvg} />, []);
+
+  const data: TargetNode = element.getData();
+  const [nodeStatus, extra] = getStatusTargetNode(data);
+
+  const classNames = React.useMemo(() => {
+    const additional = expression === '' || isTargetMatched(data, expression) ? '' : 'search-inactive';
+    return css('topology__target-node', additional);
+  }, [data, expression]);
+
+  return (
+    <Layer id={contextMenuOpen ? TOP_LAYER : DEFAULT_LAYER}>
+      {/* eslint-disable-next-line @typescript-eslint/no-explicit-any */}
+      <g className={classNames} id={'target-node-visual-group'} ref={hoverRef as any}>
```

Comment:
you can do `hoverRef as React.LegacyRef<SVGGElement>` any get rid of the eslint warning I think

Reply:
Nice thanks!